dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/token/TokenResponseValidator.java

```diff
@@ -101,8 +101,8 @@ JwtClaims getJwtClaimsFromIdTokenContext(JwtContext jwtContext) throws TokenVali
             jwtClaims = jwtContext.getJwtClaims();
         }
         if (jwtContext == null || jwtClaims == null) {
-            // TODO - NLS message
-            throw new TokenValidationException(this.clientConfig.getClientId(), "not a valid token to continue the flow");
+            String msg = Tr.formatMessage(tc, "INVALID_TOKEN_INFO", new Object[] { this.clientConfig.getClientId(), jwtContext == null ? "jwtContext" : "jwtClaims" });
+            throw new TokenValidationException(this.clientConfig.getClientId(), msg);
```

Comment:
```suggestion
            String msg = Tr.formatMessage(tc, "TOKEN_MISSING_CLAIMS", new Object[] { this.clientConfig.getClientId());
            throw new TokenValidationException(this.clientConfig.getClientId(), msg);
```
It doesn't look like it's possible for `jwtContext` to be null, so we shouldn't need to check for that here. The `jwtClaims` variable might be null, or at least I can't easily confirm that it will never be null. So I think the best approach here is to throw an exception if we find a null or empty claims object, with the NLS message saying as much.

Reply:
I updated the check and the message.